BudgetTime.java

```diff
@@ -22,24 +23,102 @@ public BudgetTime() {
 		if (numberOfDays == 0) {
 			return; //Quit if time input failed
 		}
 		int budgetAmount = 0; //Variable to hold the budget before it is stored
 		String input;
 
-		System.out.println("Please enter a budget for the selected time period");
+		System.out.println("Please enter a budget (�) for the selected time period");
 		input = userIn.readLine();
 		if (!Validation.isInteger(input)) {
 			return;
 		}
 		budgetAmount = Integer.parseInt(input);
 
-		System.out.println("Success a budget has been set for 'amount' every 'numberofdays' days!");
-		//CALL CLASS TO STORE IN A TEXT DOC OR DB
-		//ALLOW AMENDMENTS AND DELETIONS OF CURRENT BUDGETS ONCE DB IS READY
+		RetrieveAndStore.sqlExecute("INSERT INTO tblBudget (BudgetAmount, NumberOfDays) VALUES (" + budgetAmount + ", " + numberOfDays + ")");
+		//Write SQL statement here then pass to method
+		System.out.println("Success a budget has been set for �" + budgetAmount + " every " + numberOfDays + " days!");
+	}
+
+	/**
+	 * Print the budgets from the database to the user
+	 */
+	private void printBudgets() {
+		ResultSet rs = RetrieveAndStore.readAllRecords("tblBudget");
+		try {
+			while (rs.next()) //Loop through the resultset
+			{
+				//Store each budget record to print
+				int id = rs.getInt("BudgetID");
+				int budget = rs.getInt("BudgetAmount");
+				int days = rs.getInt("NumberOfDays");
+
+				// print the results
+				System.out.format("%s. Budget amount = �%s, Number of days = %s\n", id, budget, days);
+			}
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * Delete a stored budget
+	 * @throws IOException
+	 * 		Handled in main along with other IO exceptions to reduce error handling code
+	 */
+	private void deleteBudget() throws IOException {
+		printBudgets(); //Print all budgets to the console
+		System.out.println("Please enter a record number to delete");
+		String input = userIn.readLine();
+		if (!Validation.isInteger(input)) {
+			return;
+		}
+		if (!Validation.isRangeValid(1, RetrieveAndStore.maxID("tblBudget", "BudgetID"), Integer.parseInt(input))) {
+			return;
+		}
+		RetrieveAndStore.sqlExecute("DELETE FROM tblBudget WHERE BudgetID = '" + input + "'"); //Call method to execute deletion
+		System.out.format("Record %s deleted successfully\n", input); //Tell the user record has been removed		
+	}
+
+	/**
+	 * Amend a stored budget
+	 * @throws IOException
+	 * 		Handled in main along with other IO exceptions to reduce error handling code
+	 */
+	private void amendBudget() throws IOException {
+		printBudgets(); //Print all budgets to the console
+		System.out.println("Please enter a record number to amend");
+		String input = userIn.readLine();
+		if (!Validation.isInteger(input)) { //Recordnumber input validated
+			return;
+		}
+		int recordNumber = Integer.parseInt(input); 
+		if (!Validation.isRangeValid(1, RetrieveAndStore.maxID("tblBudget", "BudgetID"), recordNumber)) {
+			return;

```

Comment:
should we tell them they gave a wrong input? (here and 4 lines above)

Reply:
This is done within the validation module

Reply:
ah fair enough